Tools/debug/CollectSystemInfo.ps1

```diff
@@ -131,70 +136,47 @@ function Export-InstalledApplications {
-        Export-Csv -Path 'InstalledApplications.csv' -NoTypeInformation
+        Export-Csv -Path (Join-Path $folderPath 'InstalledApplications.csv') -NoTypeInformation
+        Write-Host 'Installed applications collection completed.'
     } catch {
         Write-Warning "Failed to collect list of installed applications: $_"
     }
 }
 
 function Export-InstalledKBs {
     try {
-        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path 'InstalledKBs.csv' -NoTypeInformation
+        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path (Join-Path $folderPath 'InstalledKBs.csv') -NoTypeInformation
+        Write-Host 'Installed KBs collection completed.'
     } catch {
         Write-Warning "Failed to collect list of installed KBs: $_"
     }
 }
 
-function Get-WindowsMemoryDump {
+function Export-WindowsMemoryDump {
     $memoryDumpPaths = @("$env:SystemRoot\MEMORY.DMP", "$env:SystemRoot\Minidump")
     
-    return $memoryDumpPaths | Where-Object { Test-Path -Path $_ }
-}
-
-
-function Get-ExportedFiles {
-    $exportedFiles = @()
-    $filesPattern = @('*.txt', '*.csv', '*.evtx', '*.log')
-    $fileInCurrentDirectory = foreach ($pattern in $filesPattern) {
-        Get-ChildItem -Path $pattern -ErrorAction SilentlyContinue
-    }
-    $exportedFiles += $fileInCurrentDirectory
-    if (Test-Path -Path 'LocaleMetaData') {
-        $exportedFiles += 'LocaleMetaData'
-    }
-
-    return $exportedFiles
-}
-
-function Remove-TempFiles {
-    param ([string[]]$Path)
-    try {
-        Remove-Item -Path $Path -Recurse -Force
-    } catch {
-        Write-Warning "Failed to delete file: $resolvedFile - $_"
+    foreach ($dump in $memoryDumpPaths) {
+        Copy-Item -Path $dump -Destination $folderPath -ErrorAction SilentlyContinue
     }
+    Write-Host 'Windows memory dump collection completed.'
 }
 
 function Write-InformationToArchive {
     try {
-        $date = Get-Date -Format 'yyyy-MM-dd_HH-mm-ss'
-        $archiveFile = "SystemInfo_$date.zip"
-        $exportedFiles = Get-ExportedFiles
-        if ($IncludeSensitiveData) {
-            $dumpFiles = Get-WindowsMemoryDump
-            $outerFiles += $dumpFiles
-        }
-        
-        $collectedFiles = $exportedFiles + $outerFiles
+        $archiveFile = "$folderName.zip"
         
-        Compress-Archive -Path $collectedFiles -DestinationPath $archiveFile
-        Remove-TempFiles $exportedFiles
+        Compress-Archive -Path $folderPath -DestinationPath $archiveFile
         Write-Host "Archiving completed ($archiveFile)."
     } catch {
         Write-Warning "Failed to archive information: $_"
     }
 }
 
+$timestamp = Get-Date -Format 'yyyy-MM-dd_HH-mm-ss'
+$folderName = "SystemInfo_$timestamp"
+$folderPath = Join-Path -Path (Get-Location) -ChildPath $folderName
+New-Item -Path $folderPath -ItemType Directory | Out-Null
```

Comment:
@YanVugenfirer 
Maybe add the log "Stating system info collecting into $folderPath"?

Reply:
Agree